src/Agoda.Analyzers/AgodaCustom/AG0018PermitOnlyCertainPubliclyExposedEnumerables.cs

```diff
@@ -15,14 +16,15 @@ public class AG0018PermitOnlyCertainPubliclyExposedEnumerables : DiagnosticAnaly
     {
         public const string DIAGNOSTIC_ID = "AG0018";
 
-        private static readonly string[] AllowedTypes = 
+        private static readonly HashSet<string> AllowedTypes = new HashSet<string>
         {
             "System.Collections.Generic.ISet<T>",
             "System.Collections.Generic.IList<T>",
             "System.Collections.Generic.IDictionary<TKey, TValue>",
             "System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>",
             "System.Collections.Generic.KeyedCollection<TKey, TValue>",
             "System.Collections.Generic.IEnumerable<T>",
+            "System.Collections.Generic.IReadOnlyCollection<T>"
```

Comment:
Could you add a test case here, please?
https://github.com/agoda-com/AgodaAnalyzers/blob/8a2214a1be4ea8e4a3eddc4626ff538e5cb51ffd/src/Agoda.Analyzers.Test/AgodaCustom/AG0018UnitTests.cs#L75-L83

Optional: if it's not a big ask, could you add `IReadOnlyList<T>` ?